api/src/main/java/uk/gov/hmcts/payment/api/controllers/PaymentGroupController.java

```diff
@@ -332,6 +361,180 @@ private Payment getPayment(PaymentFeeLink paymentFeeLink, String paymentReferenc
+        }
+
+        PaymentProvider paymentProvider = bulkScanPaymentRequestStrategic.getExternalProvider() != null ?
+            paymentProviderRepository.findByNameOrThrow(bulkScanPaymentRequestStrategic.getExternalProvider())
+            : null;
+
+        Payment payment = Payment.paymentWith()
+            .reference(referenceUtil.getNext("RC"))
+            .amount(bulkScanPaymentRequestStrategic.getAmount())
+            .caseReference(bulkScanPaymentRequestStrategic.getExceptionRecord())
+            .ccdCaseNumber(bulkScanPaymentRequestStrategic.getCcdCaseNumber())
+            .currency(bulkScanPaymentRequestStrategic.getCurrency().getCode())
+            .paymentProvider(paymentProvider)
+            .serviceType(bulkScanPaymentRequestStrategic.getService().getName())
+            .paymentMethod(PaymentMethod.
+                paymentMethodWith().name(bulkScanPaymentRequestStrategic.getPaymentMethod().getType())
+                .description("Cheque")
+                .build())
+            .paymentStatus(bulkScanPaymentRequestStrategic.getPaymentStatus())
+            .siteId(bulkScanPaymentRequestStrategic.getSiteId())
+            .giroSlipNo(bulkScanPaymentRequestStrategic.getGiroSlipNo())
+            .reportedDateOffline(DateTime.parse(bulkScanPaymentRequestStrategic.getBankedDate()).withZone(DateTimeZone.UTC).toDate())
+            .paymentChannel(bulkScanPaymentRequestStrategic.getPaymentChannel())
+            .documentControlNumber(bulkScanPaymentRequestStrategic.getDocumentControlNumber())
+            .payerName(bulkScanPaymentRequestStrategic.getPayerName())
+            .bankedDate(DateTime.parse(bulkScanPaymentRequestStrategic.getBankedDate()).withZone(DateTimeZone.UTC).toDate())
+            .build();
+
+        PaymentFeeLink paymentFeeLink = paymentGroupService.addNewPaymenttoExistingPaymentGroup(payment, paymentGroupReference);
+
+        Payment newPayment = getPayment(paymentFeeLink, payment.getReference());
+
+        // trigger Apportion based on the launch darkly feature flag
+        boolean apportionFeature = featureToggler.getBooleanValue("apportion-feature",false);
+        LOG.info("ApportionFeature Flag Value in PaymentGroupController  RecordBulkScanPaymentStrategic: {}", apportionFeature);
+        if(apportionFeature) {
+            feePayApportionService.processApportion(newPayment);
+
+            // Update Fee Amount Due as Payment Status received from Bulk Scan Payment as SUCCESS
+            if(newPayment.getPaymentStatus().getName().equalsIgnoreCase("success")) {
+                LOG.info("Update Fee Amount Due as Payment Status received from Bulk Scan Payment as SUCCESS!!!");
+                feePayApportionService.updateFeeAmountDue(newPayment);
+            }
+        }
+
+        allocateThePaymentAndMarkBulkScanPaymentAsProcessed(bulkScanPaymentRequestStrategic, paymentGroupReference, newPayment, headers);
```

Comment:
How are we regulating this feature with Launch-Darkly ?

Reply:
Front end is taking care of that

Reply:
we could have re-used the existing API & Existing request models with a backend feature-toggle for the new change,
this would save all these duplications.

Reply:
Nopes it's not a scenario of having feature toggle on and off in the future , we are not going to use old endpoint once new endpoint will be active. hence for the sake of a few weeks, we cant add the complexity of feature toggle on and off from the backend. And please see the above comments regarding the reusability of API and model and refactoring comments.

Thank you!

Reply:
If my understanding is correct, the requirement was to make a call to BSP from Payhub and update the DCN Status to Processed when PayBubble is calling PayHub to allocate a BSP Payment, so to achieve that I think we should modify the existing API with this new requirement regulated by a Feature Flag.

1. But I can't agree to this point that for an amendment to an existing feature we should create redundant Models[Inherited model should have been used] & APIs which would become a candidate for clean-up(additional effort), also how should we justify our effort to program for this clean up.
2. By propagating the Paybubble HttpHeaders to BSP through PayHub, we are sending PayBubble S2SToken to BSP in a request made by PayHub to BSP, we should send PayHub S2S Token to BSP to get the request Authenticated. I believe this is strongly against security.